src/libraries/tests.proj

```diff
@@ -388,7 +388,7 @@
     <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.IO.IsolatedStorage\tests\System.IO.IsolatedStorage.Tests.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TestNativeAot)' == 'true'">
+  <ItemGroup Condition="'$(TestNativeAot)' == 'true' and '$(RunNativeAotSmokeTestsOnly)' == 'true'">
```

Comment:
Could you make this follow the pattern that Mono team created?

* The existing switch name is `RunSmokeTestsOnly`. We can infer "NativeAOT smoke tests" from this and RunNativeAotTests whenever that's needed. It's probably only needed in one place.
* When `RunSmokeTestsOnly` is true, we start with an empty list of tests. We should add ours around the spot where Mono is doing it (https://github.com/dotnet/runtime/blob/aafe4d465f97fa3a0ae61b18c2887cf84085d25d/src/libraries/tests.proj#L410-L416). Maybe put the existing ItemGroup under `!RunNativeAotTests` and add a new one for `RunnativeAotTests==true`